clients/src/test/java/org/apache/kafka/clients/producer/internals/SenderTest.java

```diff
@@ -1154,7 +1154,9 @@ public void testExpiryOfAllSentBatchesShouldCauseUnresolvedSequences() throws Ex
         Deque<ProducerBatch> batches = accumulator.batches().get(tp0);
         assertEquals(0, batches.size());
         assertTrue(transactionManager.hasProducerId(producerId));
+
         // We should now clear the old producerId and get a new one in a single run loop.
+        time.sleep(10);
```

Comment:
Note that we blackout the node for 10ms above. Previously we were relying on the backoff logic in `maybeWaitForProducerId` for the node to be ready again. Now the test needs time to be advanced externally since we cannot rely on `client.poll` advancing it.

Reply:
SG.